app/controllers/annual_leave_requests_controller.rb

```diff
@@ -52,6 +60,15 @@ def check_leave_request_params
   end
 
   def annual_leave_request_params
-    params.require(:annual_leave_request).permit(:date_from, :date_to, :days_required, :status, :confirm_approval)
+    params.require(:annual_leave_request).permit(:date_from, :date_to, :days_required, :status, :confirm_approval, :denial_reason)
+  end
+
+  def status_update_confirmation_page
+    case annual_leave_request_params[:status]
+    when "approved"
+      confirm_annual_leave_request_approval_path
+    when "denied"
+      confirm_annual_leave_request_denial_path
+    end
```

Comment:
I agree 100%, I created a refactor [here](https://github.com/jyoung-gds/govuk-holiday-logger/pull/26) but thought it would be better as a separate PR as this one is already quite big. Let me know what you think of the refactor.

Reply:
Yeah that refactor looks like it's definitely heading in a good direction, nice one